src/main/webapp/app/components/sidebar/NavigationSidebar.tsx

```diff
@@ -1,76 +1,236 @@
+  const { isCollapsed, text, icon, nav, ...otherProps } = props;
+  const menuItem = (
+    <MenuItem icon={icon} component={nav} {...otherProps}>
+      {props.text}
+    </MenuItem>
+  );
+  if (props.isCollapsed) {
+    return (
+      <DefaultTooltip overlayInnerStyle={{ padding: '4px 5px' }} overlay={text} placement="right" mouseEnterDelay={0}>
+        {menuItem}
+      </DefaultTooltip>
+    );
+  }
+  return menuItem;
+};
+
+interface IPriorityEntityMenuItem extends MenuItemProps {
+  type: ENTITY_TYPE;
+  handlePriorityMenuItemClick: (entityType: ENTITY_TYPE) => void;
+}
+
+const PriorityEntityMenuItem = (props: IPriorityEntityMenuItem) => {
+  const { handlePriorityMenuItemClick, ...otherProps } = props;
   return (
-    <div className="sidebar-wrapper">
-      <ProSidebar collapsed={props.isSideBarCollapsed}>
-        <SidebarContent>
-          <Menu>
-            <MenuItem icon={<FontAwesomeIcon size="lg" icon={faBars} />} onClick={props.toggleSideBar} />
-          </Menu>
+    <MenuItem
+      {...otherProps}
+      component={<NavLink to={ENTITY_BASE_PATHS[props.type]} />}
+      onClick={() => {
+        props.handlePriorityMenuItemClick(props.type);
+      }}
+    />
+  );
+};
+
+export type EntityMenuFrequency = {
+  type: ENTITY_TYPE;
+  frequency: number;
+};
+
+const MenuDivider: React.FunctionComponent = () => {
+  return <div className="nav-menu-divider"></div>;
+};
+
+export const NavigationSidebar: React.FunctionComponent<StoreProps> = props => {
+  const [entityMenuFrequencies, setEntityMenuFrequencies] = useState(getDefaultEntityMenuFrequencies() as EntityMenuFrequency[]);
+  const [entityMenuOrder, setEntityMenuOrder] = useState(DEFAULT_ENTITY_MENU_ORDER);
+
+  const handlePriorityMenuItemClick = (type: ENTITY_TYPE) => {
+    if (DEFAULT_ENTITY_MENU_ORDER.includes(type)) {
+      const targetIndex = entityMenuFrequencies.findIndex(e => e.type === type);
+      entityMenuFrequencies[targetIndex] = {
+        ...entityMenuFrequencies[targetIndex],
+        frequency: entityMenuFrequencies[targetIndex].frequency + 1,
+      };
+      localStorage.setItem(PRIORITY_ENTITY_MENU_ITEM_KEY, JSON.stringify(entityMenuFrequencies));
+    }
+  };
+
+  useEffect(() => {
+    const frequencies = JSON.parse(localStorage.getItem(PRIORITY_ENTITY_MENU_ITEM_KEY));
+    if (frequencies) {
+      const parsedFrequencies = Object.keys(ENTITY_TYPE)
+        .filter((key: ENTITY_TYPE) => DEFAULT_ENTITY_MENU_ORDER.includes(ENTITY_TYPE[key]))
+        .map((key: ENTITY_TYPE) => {
+          let frequency = 0;
+          const target = _.find(frequencies, { type: ENTITY_TYPE[key] });
+          frequency = target && INTEGER_REGEX.test(target.frequency) ? parseInt(target.frequency, 10) : 0;
+          return { type: ENTITY_TYPE[key], frequency };
+        });
+      setEntityMenuFrequencies(parsedFrequencies);
+    }
+  }, []);
+
+  useEffect(() => {
+    const order = _.chain(entityMenuFrequencies)
```

Comment:
If our intention is not to recompute the frequency, do we need use the effect?

Reply:
I agree that we don't really need a second useEffect.

I made the change to move the logic to the point where the component is mounted.